File: android/app/src/main/java/com/ilass/swifthtmlwebviewapp/BridgeResponse.java
package com.ilass.swifthtmlwebviewapp;

import org.json.JSONException;
import org.json.JSONObject;

final class BridgeResponse {
    private BridgeResponse() {
    }

    static JSONObject base(JSONObject message, String action) throws JSONException {
        JSONObject response = new JSONObject();
        response.put("platform", "android");
        response.put("action", action);
        if (message != null && message.has("requestId")) {
            response.put("requestId", message.optString("requestId"));
        }
        if (message != null && message.has("paymentId")) {
            response.put("paymentId", message.optString("paymentId"));
        }
        return response;
    }

    static JSONObject error(JSONObject message, String action, String error) throws JSONException {
        JSONObject response = base(message, action);
        response.put("success", false);
        response.put("error", error);
        return response;
    }

    static JSONObject unavailable(JSONObject message, String action, String error) throws JSONException {
        JSONObject response = error(message, action, error);
        response.put("available", false);
        return response;
    }
}

File: ios/swiftHTMLWebviewApp/Models/BridgeResponse.swift
//
//  BridgeResponse.swift
//  swiftHTMLWebviewApp
//

import Foundation

enum BridgeResponse {
    static func base(request: [String: Any], action: String) -> [String: Any] {
        var response: [String: Any] = [
            "platform": "ios",
            "action": action
        ]
        if let requestId = request["requestId"] {
            response["requestId"] = requestId
        }
        if let paymentId = request["paymentId"] {
            response["paymentId"] = paymentId
        }
        return response
    }

    static func error(request: [String: Any], action: String, message: String) -> [String: Any] {
        var response = base(request: request, action: action)
        response["success"] = false
        response["error"] = message
        return response
    }

    static func unavailable(request: [String: Any], action: String, message: String) -> [String: Any] {
        var response = error(request: request, action: action, message: message)
        response["available"] = false
        return response
    }
}
